### DISCUSSION TITLE
Nuxt3 - defineNuxt* - only to specific vues

### DISCUSSION THREAD

**Discussion by nvijayap** [nuxt/nuxt#24132] [Questions]
Hi,

In Nuxt3 ...

```
  1. How to apply defineNuxtConfig only to specific (listed) .vue files?
  2. How to not apply defineNuxtConfig to specific (listed) .vue files?
  3. How to apply defineNuxtPlugin only to specific (listed) .vue files?
  4. How to not apply defineNuxtPlugin to specific (listed) .vue files?
```

Thanks

**Comment by TheAlexLichter** [comment]
* The `nuxt.config.ts` is the project-wide configuration.
* Plugins in the `plugin` directory are executed "per initial request"

Thus, both will have an effect on all files. 

What exactly is the problem you are trying to solve? 🤔

**Reply by nvijayap** [reply]
Here's what I am looking for ...

```
> file-specific runtimeConfig
```

**Reply by TheAlexLichter** [reply]
Also the runtimeConfig is app-wide.

Please tell me what issue you have and what you are trying to achieve exactly.

**Reply by nvijayap** [reply]
Alright, here's my use-case/need ...

```
  * I want all vues other than app.vue to be protected 
  * In the sense they need to redirect user to an external server for authentication
```

Currently the presence of global (config, plugin) is redirecting app.vue as well

```
  * app.vue is serving as my landing (unprotected) page
  * It should not redirect user to the external server for authentication
```

**Reply by TheAlexLichter** [reply]
What you commonly do is: add a page, call it eg login. Make this one unprotected. Then use a middleware to protect the rest 👌🏻
